### PR TITLE
Remove Transposition, refactor, and rename export functions

### PR DESCRIPTION
**Pull request by TylorS**
- Switches (temporarily) to `snabbdom-ts` from `snabbdom` in order to allow for 100% ES2015 modules throughout.
- Removes `makeHTMLDriver` because `snabbdom-to-html` does not yet have an ES2015 build. In the future this will be added back and renamed to `makeHtmlDriver`.
- Renames `makeDOMDriver` to `makeDomDriver`
- Renames `mockDOMSource` to `mockDomSource`
- Remove all traces of transposition because its a performance hit

This PR also has a tiny bit of refactoring

### CODE REVIEW

File: src/DomSources/MainDomSource.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+  dispose(): void {
+    this._isolateModule.reset();
+  }
+
+  public isolateSource: (source: DomSource, scope: string) => DomSource = isolateSource;
+  public isolateSink: (sink: Stream<VNode>, scope: string) => Stream<VNode> = isolateSink;
+}
+
+function getRootElement$(scope: string, isolateModule: IsolateModule, rootElement$: Stream<HTMLElement>) {
```
Comment (Frikki) [inline src/DomSources/MainDomSource.ts]: I find this function name and purpose confusing.

File: src/DomSources/determineUseCapture.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  `stalled`,
+  `submit`,
+  `suspend`,
+  `timeupdate`,
+  `unload`,
+  `volumechange`,
+  `waiting`,
+]
+
+export function determineUseCapture (eventType: string, options: EventsFnOptions | undefined): boolean {
```
Comment (Frikki) [inline src/DomSources/determineUseCapture.ts]: this is not a good function name that returns a boolean. How about `useCaptue`?

File: src/DomSources/select.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import { Stream } from 'most';
+import { IsolateModule } from '../modules/isolate';
+import { EventDelegator } from '../EventDelegator';
+import { DomSource } from './DomSource';
+import { MainDomSource } from './MainDomSource';
+import { ElementDomSource } from './ElementDomSource';
+
+const noop = () => void 0;
+
+function getBodyElement(): any {
```
Comment (Frikki) [inline src/DomSources/select.ts]: Why the `get` prefix?

File: src/DomSources/select.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    return document && document.body;
+  } catch (e) {
+    return {
+      addEventListener: noop,
+      removeEventListener: noop,
+    };
+  }
+}
+
+function getDocumentElement(): any {
```
Comment (Frikki) [inline src/DomSources/select.ts]: Why the `get` prefix?

File: src/DomSources/select.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    return document;
+  } catch (e) {
+    return {
+      addEventListener: noop,
+      removeEventListener: noop,
+    };
+  }
+}
+
+function getWindowElement(): any {
```
Comment (Frikki) [inline src/DomSources/select.ts]: Why the `get` prefix?

File: src/DomSources/MainDomSource.ts (hunk 1 above)
Comment (TylorS) [inline src/DomSources/MainDomSource.ts]: fixed

File: src/DomSources/determineUseCapture.ts (hunk 2 above)
Comment (TylorS) [inline src/DomSources/determineUseCapture.ts]: Then it conflicts with a number of variable names

File: src/DomSources/select.ts (hunk 4 above)
Comment (TylorS) [inline src/DomSources/select.ts]: fixed

File: src/DomSources/select.ts (hunk 5 above)
Comment (TylorS) [inline src/DomSources/select.ts]: fixed

**Pull request merged**